Use osfamily to select the correct params values

robinbowes opened the pull request:
- specifically, so this works on CentOS

derdanne commented:
Hey Rainbowes, 

thanks, i did not thought about that!

Could you please also do your changes in the spec tests, so that Travis builds will pass?

derdanne commented:
@robinbowes 

ping

robinbowes commented:
Done.

derdanne commented:
Hi @robinbowes, thanks for your work, looks pretty good! 

Sadly i have another little change request: 

Recently i changed the fail in params.pp to notices, so the module works also on other operatingsystems when all parameters are specified correctly. Could you please change this in your pull request?

robinbowes commented:
If you're going to do that then you should also set all the variables to "undef" (so your module works with strict variables turned on). Want me to add that too?

derdanne commented:
Sounds like a good idea :) Would be great if you could do that!

robinbowes commented:
OK, so I updated the params class to set parameters for unknown OSes to undef. I also noticed that changes were required on EL7 vs. EL6 (and presumably earlier) so I modified the case statement to switch on combined osfamily and major OS release version and added two separate cases for RedHat-7 and all other RedHat.

However, there's one test fail that I can't squish. I'm leaving it for now and will re-visit later, but feel free to take a look and see if you can see what's causing the failure.

derdanne commented on spec/defines/server_export_spec.rb:
```diff
@@ -3,7 +3,14 @@
+      :osfamily => 'Debian',
+      :operatingsystemmajrelease => '12.04',
+      :concat_basedir => '/dne',
+      :osfamily => 'debian',
```
This line is causing the failure - it is overwriting 'Debian' - that lowercase 'debian' is not hitting in params.pp.

derdanne commented on spec/defines/server_export_spec.rb:
```diff
@@ -3,7 +3,14 @@
+      :operatingsystemmajrelease => '12.04',
+      :concat_basedir => '/dne',
+      :osfamily => 'debian',
+      :clientcert => 'test.host',
```
I think we can remove this line.

derdanne commented:
I commented inline, i think i found the line that is causing the failure in travis ci. 

The rest looks very good, thanks for the work again!

robinbowes commented:
Aha! Thanks - I knew it would be something stupid related to the case statements not working.

I'll fix that up and push the changes later tonight.

I've got another issue relating to the /etc/exports concat resource which I'll cover in a separate ticket.

robinbowes commented:
Yay! All fixed!

derdanne commented:
Yay, thanks! 

... and merged :)
